app/mappers/grades_by_year.rb

```diff
@@ -0,0 +1,14 @@
+class GradesByYear < ROM::Mapper
+  register_as :grades_by_year
+  relation :grade_years
+
+  model GradeYear
+
+  def call(grade_years)
+    grade_years = grade_years.map{|gy| header.model.new(gy)}
+
+    grade_years.each_with_object({}) do |grade_year, hash|
+      hash[grade_year.year] = grade_year.as_json
+    end
+  end
```

Comment:
We don't have a transformation like that. I actually need that too so I'll try to figure something out. The trick is that you turn a relation into a hash, so it won't be compatible with the pipeline. It's fine to do that as the last step of the pipeline though, it's just that we don't have a DSL for it. I'd imagine something like that could be done:

``` ruby
class GradesMapper < ROM::Mapper
  step do
    attribute :id
    attribute :gpa
    attribute :year
  end

  step do
    pluck :year
  end
end
```

As far as `grade.letter` goes we don't have anything that would support that. Mappers are designed to transform arrays and hashes and here we have AR objects.